Parquet Reader: add peek()

KIwabuchi opened the pull request.

KIwabuchi committed:
Parquet Reader: add peek() & code brush up

KIwabuchi committed:
Parquet Reader: code brush up

steiltre commented:
The `peek()` function looks good. I think we should change the test though.

As it is, all ranks send the key they read or their rank ID if they did not receive a key to rank 0. I think we should send the key only if something was read. Then we can check to make sure `buf.size <= world.size()` to check if we are reading at most a single line per rank.

Commit:
Update parquet peek() test

Commit:
Update error message

Commit:
Bugfix: CMake include boost headers

KIwabuchi commented:
Thanks for reviewing!

#### 1. Changed not to send a dummy key when no row is read

I agree, we don't have to send dummy keys. I removed it.

#### 2. Changed to check `buf.size <= world.size()`

The peak API is designed to return only one row as follows:
`std::optional<std::vector<parquet_type_variant>> peek()`
So, it cannot return multiple rows by its API design. But checking  buf.size <= world.size()` would not hurt anything. I added the check.
